content/inheritance/reading/class-diagrams/_index.md

```diff
@@ -0,0 +1,90 @@
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+We have mentioned a few times the importance of designing your classes before starting to write code. It’s important to think about the fields, properties, and methods of your classes, as well as the relationship of your various classes to each other. Doing so will not only result in better code, but will also mean that you’re less likely to have to rewrite your code due to lack of foresight.
+
+**Class diagrams** are informal tools for sketching out class design. They are often discussed in the context of the [Unified Modeling Language](https://en.wikipedia.org/wiki/Unified_Modeling_Language), but most programmers use them much less formally. This will be our approach as well.
+
+We’ll introduce you to a few accepted conventions used when drawing class diagrams. These guidelines are by no means comprehensive, and they should not be thought of as rigid rules. There’s no compiler to yell at you for not drawing a class diagram properly! Instead, we encourage you to use class diagrams as planning and communication tools. Use whichever of the conventions you choose, based on your needs.
+
+Class diagrams can be created within some IDEs (including Visual Studio) and with specialized software. However, most of the time you’ll be drawing them by hand, so that’s what we’ll do here.
+
+{{% notice blue "Note" "rocket" %}} 
+
+Describing how to install and use software to create class diagrams is beyond the scope of this book. That said, drawing class diagrams by hand is still quick and immensely satisfying!
+ 
+{{% /notice %}}
+
+## Diagramming a Single Class
+
+In diagramming a single class, we create a box and divide it into three sections:
+
+   1. **Top:** Class name
+   1. **Middle:** Fields
+   1. **Bottom:** Properties and Methods
+
+For fields and properties, we can specify additional useful information, such as access level and data type. For methods, we can similarly specify access level, return type, and input parameter types. We will often use “+” as shorthand for public, and “-” as shorthand for private. If we want to indicate that some methods or fields are omitted, we can use ellipses “…”.
+
+## Diagramming Class Relationships
+
+What’s even more useful than just diagramming a single class is diagramming the relationships between the classes within our programs.
+
+Suppose we are designing a program to track students and the classes that they have taken. We might think about three classes:
+
+   1. `Student`: Represents an individual student
+   1. `Transcript`: Represents a student’s transcript, including classes that they have taken
+   1. `CourseRecord`: Represents an individual course that a student has completed, along with the grade achieved.
+
+We can represent these three classes and their relationships to each other in the following diagram:
+
+   {{< rawhtml >}}
+   <img src="../pictures/relationships.png" alt="Relationship class diagram" width=80% />
+   {{< /rawhtml >}}
```

Comment:
Picture missing, should be resolved by moving the picture folder into the reading folder.

Reply:
fixed this. needed one more'../'